compiler.go

```diff
@@ -32,17 +32,63 @@ var DebugModeTag = map[string]string{
 	"LOOP_START": "L0",
 }
 
+type BuildType string
```

Comment:
Use an Enum with a String() function, as in https://go.dev/play/p/CGj0fFeU3kJ

Reply:
we wants  string value when serializer to json, `Enum`  make it become integer when `toJSON`